client/go/cmd/cert.go

```diff
@@ -109,30 +167,91 @@ func doCert(_ *cobra.Command, args []string) error {
+		}
 	}
 
 	keyPair, err := vespa.CreateKeyPair()
 	if err != nil {
 		return err
 	}
-	pkgCertificateFile := filepath.Join(pkg.Path, "security", "clients.pem")
-	if err := os.MkdirAll(filepath.Dir(pkgCertificateFile), 0755); err != nil {
-		return fmt.Errorf("could not create security directory: %w", err)
-	}
-	if err := keyPair.WriteCertificateFile(pkgCertificateFile, overwriteCertificate); err != nil {
-		return fmt.Errorf("could not write certificate to application package: %w", err)
+	var pkgCertificateFile string
+	if !noApplicationPackage {
+		pkgCertificateFile = filepath.Join(pkg.Path, "security", "clients.pem")
+		if err := os.MkdirAll(filepath.Dir(pkgCertificateFile), 0755); err != nil {
+			return fmt.Errorf("could not create security directory: %w", err)
+		}
+		if err := keyPair.WriteCertificateFile(pkgCertificateFile, overwriteCertificate); err != nil {
+			return fmt.Errorf("could not write certificate to application package: %w", err)
+		}
 	}
 	if err := keyPair.WriteCertificateFile(certificateFile, overwriteCertificate); err != nil {
 		return fmt.Errorf("could not write certificate: %w", err)
 	}
 	if err := keyPair.WritePrivateKeyFile(privateKeyFile, overwriteCertificate); err != nil {
 		return fmt.Errorf("could not write private key: %w", err)
 	}
-	printSuccess("Certificate written to ", color.Cyan(pkgCertificateFile))
+	if !noApplicationPackage {
+		printSuccess("Certificate written to ", color.Cyan(pkgCertificateFile))
+	}
 	printSuccess("Certificate written to ", color.Cyan(certificateFile))
 	printSuccess("Private key written to ", color.Cyan(privateKeyFile))
 	return nil
 }
+
+func doCertAdd(_ *cobra.Command, args []string) error {
+	app, err := getApplication()
+	if err != nil {
+		return err
+	}
+	pkg, err := vespa.FindApplicationPackage(applicationSource(args), false)
+	if err != nil {
+		return err
+	}
+	cfg, err := LoadConfig()
+	if err != nil {
+		return err
+	}
+	certificateFile, err := cfg.CertificatePath(app)
+	if err != nil {
+		return err
+	}
+
+	if pkg.IsZip() {
+		hint := "Try running 'mvn clean' before 'vespa auth cert add', and then 'mvn package'"
+		return errHint(fmt.Errorf("unable to add certificate to compressed application package: %s", pkg.Path), hint)
+	}
+
+	pkgCertificateFile := filepath.Join(pkg.Path, "security", "clients.pem")
+	if err := os.MkdirAll(filepath.Dir(pkgCertificateFile), 0755); err != nil {
+		return fmt.Errorf("could not create security directory: %w", err)
+	}
+	src, err := os.Open(certificateFile)
+	if errors.Is(err, os.ErrNotExist) {
+		return errHint(fmt.Errorf("there is not key pair generated for application '%s'", app), "Try running 'vespa auth cert' to generate it")
+	} else if err != nil {
+		return fmt.Errorf("could not open certificate file: %w", err)
+	}
+	defer src.Close()
+	flags := os.O_CREATE | os.O_RDWR
+	if overwriteCertificate {
+		flags |= os.O_TRUNC
+	} else {
+		flags |= os.O_EXCL
+	}
```

Comment:
Are all these flags portable?

Reply:
golang rewrite them into supported windows flags:
https://github.com/golang/go/blob/master/src/syscall/syscall_windows.go#L316
All the possible combinations are supported, if the flags are changed, they may not be.